File: hw_06/CS/cs_oopwork06/cs_oopwork06/geometry/Cylinder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace cs_oopwork06
{
    public class Cylinder
    {
        private double radius;
        private double length;

        public double GetRadius()
        {
            return radius;
        }

        public void SetRadius(double radius)
        {
            this.radius = radius;
        }

        public double GetLength()
        {
            return length;
        }

        public void SetLength(double length)
        {
            this.length = length;
        }

        public Cylinder(double radius, double length)
        {
            this.radius = radius;
            this.length = length;
        }

        public Cylinder():this(1.0, 1.0){}

        public double area()
        {
            return Math.PI * Math.Pow(radius, 2) * (radius + length);
        }

        public double volume()
        {
            return area() * length;
        }

        public void CyInfo()
        {
            System.Console.WriteLine($"Cylinder: Radius = {radius:N2}, Length = {length:N2}, Area = {area():N2}, Volume = {volume():N2}");
        }
    }

}

File: hw_06/CPP/cpp_oppwork06/geometry/cylinder.cpp
#include <cmath>
#include <iostream>

using namespace std;

class Cylinder {
    private:
        double radius;
        double length;

    public:
        double getRadius(){
            return radius;
        }

        void setRadius(double radius){
            this->radius = radius;
        }

        double getLngth(){
            return length;
        }

        void setLength(double length){
            this->length = length;
        }

        Cylinder(double radius, double length): radius(radius), length(length) {}
    
        Cylinder(): Cylinder(1.0, 1.0){}

        double area(){
            return 2 * M_PI * radius * (radius + length);
        }

        double volume(){
            return area() * length;
        }

        void cyInfo(){
            cout << "Cylinder: Radius = " << radius << ", Length = " << length << ", Area = " << area() << ", Volume = " << volume() << endl;
        }

};

